src/main/java/com/cloudogu/scm/smeagol/search/IndexSyncer.java

```diff
@@ -59,6 +59,22 @@ public void ensureIndexIsUpToDate(Index<SmeagolDocument> index, Repository repos
     }
   }
 
+  public void reindex(Index<SmeagolDocument> index, Repository repository) {
+    try (RepositoryService repositoryService = repositoryServiceFactory.create(repository)) {
+      Stopwatch sw = Stopwatch.createStarted();
+      Indexer indexer = indexerFactory.create(index, repositoryService);
+      try {
+        LOG.trace("reindex started for repository {}", repositoryService.getRepository());
+        IndexSyncWorker worker = indexSyncWorkerFactory.create(repositoryService, indexer);
+        worker.reIndex();
+      } finally {
+        LOG.debug("reindex operation for repository {} finished in {}", repositoryService.getRepository(), sw.stop());
+      }
+    } catch (IOException e) {
+      LOG.error("failed reindex repository {}", repository, e);
+    }
+  }
+
```

Comment:
Maybe write tests for this method? Not sure though

Reply:
Didn't write any because `ensureIndexIsUpToDate` didn't have any either and we must have had a reason to skip it that time. The functionality is basically the same, they just call different methods on the worker.

Reply:
Okay, this is fine for me.